python/dgl/runtime/scheduler.py

```diff
@@ -74,25 +75,37 @@ def schedule_recv(graph,
     inplace: bool
         If True, the update will be done in place
     """
-    src, dst, mid = graph._msg_graph.in_edges(recv_nodes)
-    if len(mid) == 0:
-        # All recv nodes are 0-degree nodes; downgrade to apply nodes.
+    _, _, eid = graph._graph.in_edges(recv_nodes)
+    if len(eid) > 0:
+        e_mask = graph._msg_index.get_items(eid).nonzero()
+        eid = eid.get_items(e_mask)
+    if len(eid) == 0:
+        # Downgrade to apply nodes if
+        #   1) all recv nodes are 0-degree nodes
+        #   2) no send has been called
         if apply_func is not None:
             schedule_apply_nodes(graph, recv_nodes, apply_func, inplace)
     else:
+        src, dst, _ = graph._graph.find_edges(eid)
```

Comment:
`find_edges` can be avoided? Using mask on the src and dst in L78.

Reply:
Maybe, only if line 80 `graph._msg_index.get_items(eid).nonzero()` returns all edges, otherwise eid will be changed.